packages/plugin/vite/src/util/package.ts

```diff
@@ -0,0 +1,116 @@
+import fs from 'node:fs';
+import path from 'node:path';
+
+import type { PackageJsonManifest } from './packageJson';
+
+export interface Dependency {
+  name: string;
+  path: SourceAndDestination;
+  dependencies: Dependency[];
+}
+
+export interface SourceAndDestination {
+  src: string;
+  dest: string;
+}
+
+function isRootPath(dir: string) {
+  // Linux or Windows root path
+  return dir === '/' || /^[A-Z]:\\$/i.test(dir);
+}
+
+export async function lookupNodeModulesPaths(root: string, paths: string[] = []): Promise<string[]> {
+  if (!root) return paths;
+  if (!path.isAbsolute(root)) return paths;
+
+  const p = path.join(root, 'node_modules');
+
+  if (fs.existsSync(p) && (await fs.promises.stat(p)).isDirectory()) {
```

Comment:
```ts
export const isDirectory = (p: string): Promise<boolean> => {
  try {
    const stat = fs.promise.stat(p);
    return stat.isDirectory();
  } catch(_err: Error) {
    return false;
  }
}
```

We can encapsulate a function for this to avoid calling: `*Sync` methods.